cli/args/mod.rs

```diff
@@ -1792,54 +1759,54 @@ fn resolve_node_modules_folder(
   cwd: &Path,
   flags: &Flags,
   workspace: &Workspace,
-  maybe_config_file: Option<&ConfigFile>,
-  maybe_package_json: Option<&PackageJson>,
   deno_dir_provider: &Arc<DenoDirProvider>,
 ) -> Result<Option<PathBuf>, AnyError> {
-  let use_node_modules_dir = if *DENO_FUTURE {
-    if let Some(mode) = flags.node_modules_mode {
-      Some(mode.uses_node_modules_dir())
-    } else {
-      workspace
-        .node_modules_mode()?
-        .map(|m| m.uses_node_modules_dir())
-        .or(flags.vendor)
-        .or_else(|| maybe_config_file.and_then(|c| c.json.vendor))
-    }
+  fn resolve_from_root(root_folder: &FolderConfigs, cwd: &Path) -> PathBuf {
+    root_folder
+      .deno_json
+      .as_ref()
+      .map(|c| Cow::Owned(c.dir_path()))
+      .or_else(|| {
+        root_folder
+          .pkg_json
+          .as_ref()
+          .map(|c| Cow::Borrowed(c.dir_path()))
+      })
+      .unwrap_or(Cow::Borrowed(cwd))
+      .join("node_modules")
+  }
+
+  let root_folder = workspace.root_folder_configs();
+  let use_node_modules_dir = if let Some(mode) = flags.node_modules_dir {
+    Some(mode.uses_node_modules_dir())
   } else {
-    flags
-      .node_modules_dir
-      .or_else(|| maybe_config_file.and_then(|c| c.json.node_modules_dir))
+    workspace
+      .node_modules_dir_mode()?
+      .map(|m| m.uses_node_modules_dir())
       .or(flags.vendor)
-      .or_else(|| maybe_config_file.and_then(|c| c.json.vendor))
+      .or_else(|| root_folder.deno_json.as_ref().and_then(|c| c.json.vendor))
   };
   let path = if use_node_modules_dir == Some(false) {
     return Ok(None);
   } else if let Some(state) = &*NPM_PROCESS_STATE {
     return Ok(state.local_node_modules_path.as_ref().map(PathBuf::from));
-  } else if let Some(package_json_path) = maybe_package_json.map(|c| &c.path) {
+  } else if workspace.package_jsons().next().is_some() {
```

Comment:
Fixed this to have the same behaviour no matter where the package.json is in the workspace (this was an oversight previously).